src/Service/HaalCentraalToStufBGService.php

```diff
@@ -38,21 +54,119 @@ class HaalCentraalToStufBGService
 
 
     /**
-     * @param EntityManagerInterface $entityManager The Entity Manager.
-     * @param LoggerInterface        $pluginLogger  The plugin version of the logger interface.
+     * @param CallService            $callService            The CallService.
+     * @param MappingService         $mappingService         The MappingService.
+     * @param GatewayResourceService $gatewayResourceService The GatewayResourceService.
+     * @param LoggerInterface        $pluginLogger           The plugin version of the logger interface.
      */
     public function __construct(
-        EntityManagerInterface $entityManager,
+        CallService $callService,
+        MappingService $mappingService,
+        GatewayResourceService $gatewayResourceService,
         LoggerInterface $pluginLogger
     ) {
-        $this->entityManager = $entityManager;
-        $this->logger        = $pluginLogger;
-        $this->configuration = [];
-        $this->data          = [];
+        $this->callService            = $callService;
+        $this->mappingService         = $mappingService;
+        $this->gatewayResourceService = $gatewayResourceService;
+        $this->logger                 = $pluginLogger;
+        $this->configuration          = [];
+        $this->data                   = [];
 
     }//end __construct()
 
 
+    /**
+     * Finds source by reference.
+     *
+     * @return Source|null The resulting source.
+     */
+    public function getSource(): ?Source
+    {
+        $source = $this->gatewayResourceService->getSource('https://commongateway.nl/source/stuf.haalcentraal.source.json', 'haalcentraal-to-stufbg-bundle');
+        if ($source instanceof Source === false) {
+            $this->logger->error("No source found with reference: https://commongateway.nl/source/stuf.haalcentraal.source.json");
+
+            return null;
+        }
+
+        return $source;
+
+    }//end getSource()
+
+
+    /**
+     * Finds mapping by reference.
+     *
+     * @return Mapping|null The resulting mapping.
+     */
+    public function getMapping(): ?Mapping
```

Comment:
Het idee van de $this->gatewayResourceService->getMapping functie is dat je dit niet meer hoeft te doen